runtime/lua/vim/lsp.lua

```diff
@@ -1112,19 +1112,43 @@ function lsp.start_client(config)
     end
   end
 
+  ---@private
+  -- Determines whether the given option can be set by `set_defaults`.
+  local function is_empty_or_default(bufnr, option)
+    if vim.bo[bufnr][option] == '' then
+      return true
+    end
+
+    local old_bufnr = vim.fn.bufnr('')
+    local last_set_from = vim.fn.gettext('\n\tLast set from ')
+    local line = vim.fn.gettext(' line ')
+
+    vim.cmd.buffer(bufnr)
+    local scriptname = vim.fn
+      .execute('verbose set ' .. option .. '?')
+      :match(last_set_from .. '(.*)' .. line .. '%d+')
+    vim.cmd.buffer(old_bufnr)
```

Comment:
Should I send a PR with this change?

Reply:
Well I guess this depends on what is causing the buffer to be deleted. @Ursescu would you be able to build and test if I open a PR? Or alternatively could you please share a minimal vimrc and steps needed to reproduce the problem?

Reply:
I copy-pasted the `noautocmd` variant, but it doesn't work... I am not sure why, because this made sense. I will investigate further and get back to you.

Reply:
After doing some tests I concluded that the command itself will cause the NvimTree buffer to be unloaded. I am not sure how `:buffer` really works under the hood and what interactions are between unlisted buffers and normal buffers. Even if I wrap the whole thing with `eventignore` to stop the events (I actually added prints on each autocmd registered and none is triggered), still the buffer is unloaded. I tried the command `:buffer 4` (some random existing buffer) while current buffer == NvimTree buffer and I see the same behavior (buffer unloaded).
```
:buffers!                                                                                                                                                                                                           
  2u h   "[Scratch]"                    line 0                                                                                                                                                                      
  3  a   "init.lua"                     line 175                                                                                                                                                                    
  4      "stuff.txt"                    line 1                                                                                                                                                                      
  5      "lua/autocmds.lua"             line 27                                                                                                                                                                     
  6 #a   "~/.config/nvim/lua/plugins.lua" line 101                                                                                                                                                                  
  8u%a-  "NvimTree_1"                   line 4  
  ```
  So in the end I think the issue is more or less dependent on the setup/plugins/configuration and not necessarily due to this PR. Still it would be awesome to not rely on buffer switching here.

Reply:
try `:verbose 15` to see what autocmds are being fired as they happen, might help with narrowing it down